File: CPP/No888.cc
/**
 * Created by Xiaozhong on 2020/12/22.
 * Copyright (c) 2020/12/22 Xiaozhong. All rights reserved.
 */
#include <vector>
#include <iostream>
#include <unordered_set>

using namespace std;

class Solution {
public:
    vector<int> fairCandySwap(vector<int> &A, vector<int> &B) {
        int sumA = 0, sumB = 0;
        for (int i : A) sumA += i;
        for (int i : B) sumB += i;
        int delta = (sumA - sumB) / 2;
        unordered_set<int> set;
        for (int i : A) set.insert(i);
        for (int i : B) {
            if (set.find(i + delta) != set.end()) {
                return {i + delta, i};
            }
        }
        return {};
    }
};

File: Java/src/No888.java
import java.util.HashSet;

/**
 * Created by Xiaozhong on 2021/2/1.
 * Copyright (c) 2021/2/1 Xiaozhong. All rights reserved.
 */

public class No888 {
    public int[] fairCandySwap(int[] A, int[] B) {
        int sumA = 0, sumB = 0;
        for (int i : A) sumA += i;
        for (int i : B) sumB += i;
        int delta = (sumA - sumB) / 2;
        HashSet<Integer> set = new HashSet<>();
        for (int i : A) set.add(i);
        for (int i : B) {
            if (set.contains(i + delta)) return new int[]{i + delta, i};
        }
        return null;
    }
}
